### ISSUE TITLE
ipython not building for python 3.3, 3.4

### ISSUE THREAD

**Issue by lheckemann** [NixOS/nixpkgs#23839]
## Issue description
ipython only builds with python 3.5 or 2.7. Reason is unclear, [here's the log](https://github.com/NixOS/nixpkgs/files/838231/log.txt).

### Steps to reproduce
```
nix-build --no-out-link -E 'with import <nixpkgs> {}; python33.withPackages (ps: with ps; [ipython])'
nix-build --no-out-link -E 'with import <nixpkgs> {}; python34.withPackages (ps: with ps; [ipython])'
```

## Technical details

* System: 17.03beta320.ade5837 (Gorilla)
* Nix version: 1.11.7
* Nixpkgs version: "17.03beta320.ade5837"

**Comment by FRidh** [comment]
Thanks for reporting. I've pushed some updates and fixes to `release-17.03`. `ipython` now builds on Python 3.4. 

On Python 3.3 the following issue remains https://github.com/HypothesisWorks/hypothesis-python/issues/477 as well as `pathlib2` that doesn't build.

**Comment by lheckemann** [comment]
Great, thanks!
